base_layer/wallet/src/transaction_service/service.rs

```diff
@@ -4139,6 +4156,270 @@ where
+            },
+            TransactionDirection::Unknown => {
+                // Fallback to all outputs if direction is unknown
+                let all_outputs: Vec<FixedHash> = original_transaction
+                    .transaction
+                    .body
+                    .outputs()
+                    .iter()
+                    .map(|output| output.hash())
+                    .collect();
+                debug!(
+                    target: LOG_TARGET,
+                    "user-pay-for-fee: Unknown transaction direction - using all {} outputs",
+                    all_outputs.len()
+                );
+                all_outputs
+            },
+        };
+
+        // Convert filtered hashes to CompressedCommitments by finding matching outputs
+        let original_outputs: Vec<CompressedCommitment> = original_transaction
+            .transaction
+            .body
+            .outputs()
+            .iter()
+            .filter(|output| filtered_output_hashes.contains(&output.hash()))
+            .map(|output| output.commitment().to_owned())
+            .collect();
+
+        // Calculate total amount from original outputs by decrypting their encrypted data
+        let mut total_amount = MicroMinotari::zero();
+        let view_key = match self
+            .resources
+            .transaction_key_manager_service
+            .get_private_view_key()
+            .await
+        {
+            Ok(key) => key,
+            Err(e) => {
+                warn!(
+                    target: LOG_TARGET,
+                    "user-pay-for-fee: Could not get private view key: {}",
+                    e
+                );
+                return Err(TransactionServiceError::KeyManagerServiceError(e));
+            },
+        };
+
+        // Find the actual outputs from the transaction and decrypt their amounts
+        for commitment in &original_outputs {
+            if let Some(output) = original_transaction
+                .transaction
+                .body
+                .outputs()
+                .iter()
+                .find(|output| output.commitment() == commitment)
+            {
+                match EncryptedData::decrypt_data(&view_key, output.commitment(), output.encrypted_data()) {
+                    Ok((amount, _, _)) => {
+                        total_amount += amount;
+                    },
+                    Err(e) => {
+                        warn!(
+                            target: LOG_TARGET,
+                            "user-pay-for-fee: Could not decrypt amount for commitment {:?}: {}",
+                            commitment,
+                            e
+                        );
+                    },
+                }
+            } else {
+                warn!(
+                    target: LOG_TARGET,
+                    "user-pay-for-fee: Could not find transaction output for commitment {:?}",
+                    commitment
+                );
+            }
+        }
```

Comment:
Consider this simplification, as only outputs that can be decrypted are owned by this wallet.
```suggestion
let all_outputs = original_transaction
    .transaction
    .body
    .outputs()
    .iter()
    .collect::<Vec<_>>();

let mut spendable_outputs = Vec::new();
let mut total_amount = MicroMinotari::zero();

let view_key = self.resources.transaction_key_manager_service.get_private_view_key().await?;

for output in all_outputs {
    match EncryptedData::decrypt_data(&view_key, output.commitment(), output.encrypted_data()) {
        Ok((amount, _, _)) => {
            total_amount += amount;
            spendable_outputs.push(output.commitment().clone());
        },
        Err(_) => {
            debug!(target: LOG_TARGET, "Output {:?} not decryptable; skipping.", output.commitment());
        }
    }
}
```
Then later on
```rust
UtxoSelectionCriteria::must_include(spendable_outputs)
```

Reply:
This is a great way for getting spendable outputs. Change code as suggested to simplify logic.